pkg/kube/client.go

```diff
@@ -739,20 +748,38 @@ func (c *Client) pollCRDUntilEstablished(timeout time.Duration, info *resource.I
 			return false, fmt.Errorf("unable to get CRD: %v", err)
 		}
 
-		crd := &apiextv1beta1.CustomResourceDefinition{}
-		err = scheme.Scheme.Convert(info.Object, crd, nil)
-		if err != nil {
-			return false, fmt.Errorf("unable to convert to CRD type: %v", err)
+		crd, ok := info.Object.(*unstructured.Unstructured)
+		if !ok {
+			return false, fmt.Errorf("unable to cast CRD to unstructured.Unstructured type: %v", err)
+		}
+
+		conditions, ok, err := unstructured.NestedSlice(crd.UnstructuredContent(), "status", "conditions")
```

Comment:
Why are we casting to an unstructured object, then trying to determine if `.status.conditions` is part of the object? Why can't we do the same conversion logic as is available in Helm 3?

In other words..


```
switch value := AsVersioned(info.Object).(type) {
case *apiextv1beta1.CustomResourceDefinition:
    ...
case *apiextv1.CustomResourceDefinition:
    ...
}
```

There may be other object types that also report their status via the `.status.conditions` field. As I understand it, converting this object to an unstructured type means we're no longer checking the object's type, and could potentially trigger wait logic on a non-CRD type (like a Deployment).